src/runtime/propensity-server.js

```diff
@@ -103,11 +103,18 @@ export class PropensityServer {
       method: 'GET',
       credentials: 'include',
     });
-    let userState = this.publicationId_ + ':' + state;
+    let stateParam = this.publicationId_ + ':' + state;
+    let extrainfoParam = '&extrainfo=';
     if (productsOrSkus) {
-      userState = userState + ':' + encodeURIComponent(productsOrSkus);
+      const skus = encodeURIComponent(productsOrSkus);
+      extrainfoParam += skus;
+      //TODO: Remove this once propensity officially moves over to the new param
+      stateParam += ':' + skus;
     }
-    const url = adsUrl('/subopt/data?states=') + encodeURIComponent(userState);
+    const url =
+      adsUrl('/subopt/data?states=') +
+      encodeURIComponent(stateParam) +
```

Comment:
Could you use the `addQueryParam` util method to add the query params? It handles URL encoding, checking whether `?` or `&` is needed, etc

Reply:
Done